contracts/strategy/LiquidationStrategy.sol

```diff
@@ -144,80 +162,151 @@ contract LiquidationStrategy is BaseStrategy, ReentrancyGuard, IFlashLendingCall
     /// @param _router The UniswapV2Router02 (or a fork) used for the sale.
-    /// @param _amount The amount of WXDC to sell.
+    /// @param _amount The amount of Collateral to sell.
     /// @param _minAmountOut The minimum amount of FXD to accept for the sale.
-    function sellWXDC(IUniswapV2Router02 _router, uint256 _amount, uint256 _minAmountOut) external onlyStrategyManager {
-        require(address(_router) != address(0), "LiquidationStrategy: zero address");
-        require(_amount > 0, "LiquidationStrategy: zero amount");
-        require(_amount <= idleWXDC.WXDCAmount, "LiquidationStrategy: wrong amount");
+    function sellCollateralV2(address _collateral, IUniswapV2Router02 _router, uint256 _amount, uint256 _minAmountOut) external onlyStrategyManager {
+        if (address(_router) == address(0)) revert ZeroAddress();
+        if (_amount == 0) revert ZeroAmount();
+        if (_amount > idleCollateral[_collateral].collateralAmount) revert WrongAmount();
+
+        // Calculate the cost of Collateral in terms of FXD using the average price
+        uint256 _averagePriceOfCollateral = idleCollateral[_collateral].averagePriceOfCollateral;
+        uint256 costOfCollateralInFXD = _amount.mul(_averagePriceOfCollateral).div(WAD); 
+        uint256 balanceOfFXDBeforeSwap = fathomStablecoin.balanceOf(address(this));
+        
+        // Existing logic for selling Collateral
+        idleCollateral[_collateral].collateralAmount -= _amount;
+
+        _nullifyIdleCollateral(_collateral);
+
+        //if idleCollateral is nullified in previous line, no need to _adjustAmountNeededToPayDebt
+        if (idleCollateral[_collateral].collateralAmount != 0) {
+            _adjustAmountNeededToPayDebt(_collateral, _amount.mul(_averagePriceOfCollateral).div(WAD));
+        }
 
-        idleWXDC.WXDCAmount -= _amount;
+        (address[] memory path, uint256 dexAmountOut) = _computeMostProfitablePath(_router, _collateral, _amount);
 
-        if (idleWXDC.WXDCAmount == 0) {
-            idleWXDC.amountNeededToPayDebt = 0;
-            idleWXDC.averagePriceOfWXDC = 0;
-        }
+        if (_minAmountOut > dexAmountOut) revert DEXCannotGiveEnoughAmount();
 
-        uint256 deductAmountNeededToPayDebt = _amount.mul(idleWXDC.averagePriceOfWXDC).div(WAD);
-        if (idleWXDC.amountNeededToPayDebt >= deductAmountNeededToPayDebt) {
-            idleWXDC.amountNeededToPayDebt -= deductAmountNeededToPayDebt;
-        } else {
-            idleWXDC.amountNeededToPayDebt = 0;
+        uint256 receivedAmount = _sellCollateralV2(_collateral, path, _router, _amount, _minAmountOut);
+        
+        emit LogSellCollateralV2(path, _router, _amount, _minAmountOut, dexAmountOut, receivedAmount);
+
+        uint256 balanceOfFXDAfterSwap = fathomStablecoin.balanceOf(address(this));
+        uint256 fxdReceived = balanceOfFXDAfterSwap.sub(balanceOfFXDBeforeSwap); // FXD gained from the swap
```

Comment:
here, we have the same double-calculation issue as described in the comment to the `sellCollateralV3` fn.
`receivedAmount` is already calculated as the difference between the balance before and after the swap. and separate calculations for `fxdReceived` are redundant